app/labor/markdown_traverser.rb

```diff
@@ -0,0 +1,35 @@
+# To go through a markdown document. Mainly used to decide if we are in
+# code blocks to decide if we should escape underscored usernames.
+class MarkdownTraverser
+  def initialize(markdown)
+    @lines = markdown.dup.lines
+    init_position
+  end
+
+  def each
+    lines.each do |line|
+      update_position(line.include?(CODEBLOCK_MARKER))
+      yield(line)
+    end
+  end
+
+  def in_codeblock?
+    prev == true && current == false
+  end
+
+  private
+
+  CODEBLOCK_MARKER = "```".freeze
```

Comment:
Not that the visibility of this matters much, just checking in to see if you're aware that moving constants in a `private` block will not actually make them private, you need the `private_constant` method for that.

Reply:
Yup 👍